src/txn/txn_rollback_to_stable.c

```diff
@@ -1375,35 +1379,27 @@ __txn_user_active(WT_SESSION_IMPL *session)
             txn_active = true;
             break;
         }
+
+        /* Check if a user session has an active file cursor. */
+        if (session_in_list->ncursors != 0) {
+            cursor_active = true;
+            break;
+        }
     }
     __wt_spin_unlock(session, &conn->api_lock);
 
     /*
-     * A new transaction may start after we return from this call and callers should be aware of
-     * this limitation.
-     */
-    return (txn_active);
-}
-
-/*
- * __rollback_to_stable_check --
- *     Ensure the rollback request is reasonable.
- */
-static int
-__rollback_to_stable_check(WT_SESSION_IMPL *session)
-{
-    WT_DECL_RET;
-
-    /*
-     * Help the user comply with the requirement that there are no concurrent user operations. It is
-     * okay to have a transaction in prepared state.
+     * A new cursor may be positioned or a transaction may start after we return from this call and
+     * callers should be aware of this limitation.
      */
-    if (__txn_user_active(session)) {
+    if (cursor_active)
+        WT_RET_MSG(session, EBUSY, "rollback_to_stable illegal with active file cursors");
+    if (txn_active) {
+        ret = EBUSY;
         WT_TRET(__wt_verbose_dump_txn(session));
-        WT_RET_MSG(session, EBUSY, "rollback_to_stable illegal with active transactions");
+        WT_RET_MSG(session, ret, "rollback_to_stable illegal with active transactions");
```

Comment:
Why don't you just directly pass EBUSY here? Looks like it'll override any previously set ret. You could also `WT_IGNORE_RET` the verbose dump txn call.

Reply:
Check out `WT_TRET`, generally the first error is retained and not overridden. (And, I wouldn't want EBUSY to override WT_PANIC, for example.) I wouldn't be too upset if `__wt_verbose_dump_txn()` stopped returning error values at all, but as long as it does return them, I don't like ignoring them, I think it's bad practice.

Reply:
So we'd be capturing `WT_PANIC` or other more severe errors than just `EBUSY`. I think that's fine.